File: backend/src/main/java/com/ssafy/cheket/service/resale/ResaleService.java
package com.ssafy.cheket.service.resale;

import com.ssafy.cheket.dto.resale.response.GetResaleTicketsResponse;
import com.ssafy.cheket.dto.resale.response.ResaleShowItem;
import com.ssafy.cheket.dto.show.response.GetShowListResponse;
import com.ssafy.cheket.enums.ResaleShowSort;
import com.ssafy.cheket.enums.ResaleTicketSort;

import java.util.List;

public interface ResaleService {
    GetShowListResponse<ResaleShowItem> getResaleShowList(List<Integer> regions, ResaleShowSort sort, String keyword,
        int page, int size);

    GetResaleTicketsResponse getResaleTickets(Long showId, ResaleTicketSort sort, Long sessionId, int page, int size);

    // 리세일 취소
    Long cancelResale(Long userId, Long ticketId);

    // 리세일 구매
    Long purchaseResale(Long userId, Long ticketId);
}

File: frontend/apps/customer-app/android/app/src/main/java/com/ssafy/cheket/core/network/service/ResaleService.kt
package com.ssafy.cheket.core.network.service

import com.ssafy.cheket.core.network.dto.*
import retrofit2.http.*

interface ResaleService {

    @POST("api/v1/tickets/{ticketId}/resales")
    suspend fun registerResale(
        @Path("ticketId") ticketId: Long,
        @Body request: ResaleRegisterRequest,
    ): ApiResponse<TxIdResponse>

    @GET("api/v1/resales")
    suspend fun getResaleShows(
        @Query("regions") regions: List<Int>? = null,
        @Query("sort") sort: String? = null,
        @Query("keyword") keyword: String? = null,
        @Query("page") page: Int = 0,
        @Query("size") size: Int = 20,
    ): ApiResponse<ResaleShowListResponse>

    @GET("api/v1/resales/shows/{showId}")
    suspend fun getResaleTickets(
        @Path("showId") showId: Long,
        @Query("sort") sort: String? = null,
        @Query("sessionId") sessionId: Long? = null,
        @Query("page") page: Int = 0,
        @Query("size") size: Int = 20,
    ): ApiResponse<ResaleTicketListResponse>

    @POST("api/v1/resales/{ticketId}")
    suspend fun purchaseResale(@Path("ticketId") ticketId: Long): ApiResponse<TxIdResponse>

    @DELETE("api/v1/resales/{ticketId}")
    suspend fun cancelResale(@Path("ticketId") ticketId: Long): ApiResponse<TxIdResponse>
}
